### PR TITLE
Switch to using vendor directory for deps.

### PR DESCRIPTION
**Pull request by vishh**
Fixes https://github.com/google/cadvisor/issues/1181

### CODE REVIEW

**Comment by vishh** [comment]
cc @timstclair

File: vendor/github.com/Shopify/sarama/.travis.yml [hunk 1]
```diff
@@ -1,7 +1,7 @@
 language: go
 go:
 - 1.4.3
-- 1.5.2
```
Comment (timstclair) [inline vendor/github.com/Shopify/sarama/.travis.yml]: Why did `Shopify/sarama` files change? I don't see any corresponding changes in `Godeps.json`

File: vendor/github.com/abbot/go-http-auth/digest.go [hunk 2]
```diff
@@ -22,7 +22,7 @@ type DigestAuth struct {
 	Secrets          SecretProvider
 	PlainTextSecrets bool
 
-	/*
```
Comment (timstclair) [inline vendor/github.com/abbot/go-http-auth/digest.go]: ditto RE: misc changes

File: docs/development/build.md [hunk 3]
```diff
@@ -14,24 +14,18 @@ After setting up Go, you should be able to `go get` cAdvisor as expected (we use
 $ go get -d github.com/google/cadvisor
```
Comment (timstclair) [inline docs/development/build.md]: There's conflicting information above: "Ensure that your version of Go is at least 1.3." and "**Note**: cAdvisor requires Go 1.5 to build.". Could you fix it? Does the vendor directory work by default on go 1.5 or do you need to set a flag?

File: build/presubmit.sh [hunk 4]
```diff
@@ -19,6 +19,4 @@ set -x
 
 ./build/check_gofmt.sh .
 ./build/check_boilerplate.sh
-go vet github.com/google/cadvisor/...
-godep go test -v -race -test.short github.com/google/cadvisor/...
-godep go build github.com/google/cadvisor
+make
```
Comment (timstclair) [inline build/presubmit.sh]: how about instead of putting this here, we add a presubmit make rule that depends on the necessary other rules? (then change the presubmit call in the jenkins script to call make instead)

Comment (vishh) [inline build/presubmit.sh]: Great idea. Can we do that in a separate PR?

File: deploy/build.sh [hunk 5]
```diff
@@ -17,6 +17,6 @@
 set -e
 set -x
 
-godep go build -a github.com/google/cadvisor
+make
```
Comment (timstclair) [inline deploy/build.sh]: I'd prefer `make build` here. I don't think `fmt` `vet` and `test` need to be part of this script.

File: docs/development/build.md (hunk 3 above)
Comment (vishh) [inline docs/development/build.md]: Good catch. Fixing it. I'm going to make go 1.6 minimum requirement.

File: deploy/canary/Dockerfile [hunk 6]
```diff
@@ -3,8 +3,7 @@ MAINTAINER vmarmol@google.com
 
 RUN apt-get install -y git dmsetup
 RUN git clone https://github.com/google/cadvisor.git /go/src/github.com/google/cadvisor
-RUN go get github.com/tools/godep
-RUN cd /go/src/github.com/google/cadvisor && godep go build .
+RUN cd /go/src/github.com/google/cadvisor && make
```
Comment (timstclair) [inline deploy/canary/Dockerfile]: ditto: use `make build` here. Are you sure the Makefile should be the source of truth for all our build commands? I'm inclined to prefer calling `build/build.sh` directly, but don't have any concrete reasons for it.

File: deploy/canary/Dockerfile [hunk 7]
```diff
@@ -3,8 +3,7 @@ MAINTAINER vmarmol@google.com
 
 RUN apt-get install -y git dmsetup
```
Comment (timstclair) [inline deploy/canary/Dockerfile]: Can we just delete this file?  `make docker` seems sufficient for local development and uses `deploy/Dockerfile` instead.

File: docs/development/build.md [hunk 8]
```diff
@@ -50,6 +44,6 @@ If you modify files in the /assets folder, you will need to rebuild the assets:
 
 ```
 $GOPATH/src/github.com/google/cadvisor $ ./build/assets.sh
-$GOPATH/src/github.com/google/cadvisor $ godep go build .
+$GOPATH/src/github.com/google/cadvisor $ make
```
Comment (timstclair) [inline docs/development/build.md]: `make build` includes ./build/assets.sh, so the above line is redundant. I've been meaning to split out a separate `make assets` rule that build depends on...

File: docs/development/build.md [hunk 9]
```diff
@@ -14,24 +14,18 @@ After setting up Go, you should be able to `go get` cAdvisor as expected (we use
...
-$ go get github.com/tools/godep
-```
-
 ## Building from Source
 
 At this point you can build cAdvisor from the source folder:
 
 ```
-$GOPATH/src/github.com/google/cadvisor $ godep go build .
+$GOPATH/src/github.com/google/cadvisor $ make
```
Comment (timstclair) [inline docs/development/build.md]: Again, I'd prefer to be explicit about which rule to execute (do we really need `all` here?)

**Comment by timstclair** [comment]
Thanks for fixing this!

File: build/presubmit.sh (hunk 4 above)
Comment (timstclair) [inline build/presubmit.sh]: As long as you promise to do it :) I'm worried that leaving all these naked `make` calls around is fragile.

File: vendor/github.com/Shopify/sarama/.travis.yml (hunk 1 above)
Comment (vishh) [inline vendor/github.com/Shopify/sarama/.travis.yml]: Honestly, I have no idea how I managed to pull in a newer version of this dep. Do you think its worth fixing?

File: docs/development/build.md (hunk 3 above)
Comment (vishh) [inline docs/development/build.md]: Fixing it.

File: deploy/canary/Dockerfile (hunk 7 above)
Comment (vishh) [inline deploy/canary/Dockerfile]: This is for our jenkins automation that builds a canary docker image from HEAD every night. We can update it I guess to use existing make rules. Again can we do that in a separate PR?

File: docs/development/build.md (hunk 8 above)
Comment (vishh) [inline docs/development/build.md]: Ack. Removing this section.

File: docs/development/build.md (hunk 9 above)
Comment (vishh) [inline docs/development/build.md]: Depends on the context I think. We definitely want contributors to run `make all` before posting patches. But it is not necessary for building though.

File: deploy/canary/Dockerfile (hunk 6 above)
Comment (vishh) [inline deploy/canary/Dockerfile]: I will clean this up in a separate PR since it needs some work on the jenkins side as well

File: build/presubmit.sh (hunk 4 above)
Comment (vishh) [inline build/presubmit.sh]: Ok. I'm doing it in this patch itself :)

**Comment by vishh** [comment]
@timstclair posted a new patch. PTAL

File: deploy/canary/Dockerfile (hunk 6 above)
Comment (vishh) [inline deploy/canary/Dockerfile]: https://github.com/google/cadvisor/pull/1372

File: deploy/canary/Dockerfile (hunk 7 above)
Comment (timstclair) [inline deploy/canary/Dockerfile]: Ah, that makes sense, sorry I missed that before. Do you think it's better to build cAdvisor in the docker container, like it was before? Or does it not make a difference? I'm OK with whichever you think is best :)

File: vendor/github.com/Shopify/sarama/.travis.yml (hunk 1 above)
Comment (timstclair) [inline vendor/github.com/Shopify/sarama/.travis.yml]: I was able to reproduce this in the old tree with:
```
godep restore
rm -rf Godeps
export  GO15VENDOREXPERIMENT=0
godep save ./...
```
I filed https://github.com/google/cadvisor/issues/1373 to prevent this in the future, but I think you can leave this as is.

File: vendor/github.com/abbot/go-http-auth/digest.go (hunk 2 above)
Comment (timstclair) [inline vendor/github.com/abbot/go-http-auth/digest.go]: <ignore>

**Comment by timstclair** [comment]
LGTM. Go ahead and self-merge after squashing the comment commits.
